api/nodes/serializers.py

```diff
@@ -1094,6 +1095,11 @@ def validate_data(self, node, user_id=None, full_name=None, email=None, index=No
             raise exceptions.ValidationError(detail='A user ID or full name must be provided to add a contributor.')
         if user_id and email:
             raise exceptions.ValidationError(detail='Do not provide an email when providing this user_id.')
+        if email:
+            try:
+                validate_email(email)
+            except BlacklistedEmailError:
+                raise exceptions.ValidationError(detail='Invalid email for unregistered contributor')
```

Comment:
Small - wherever you end up throwing your exception, be sure you have punctuation, and I'd add a little more detail - maybe here, that the issue is that it is an invalid email domain.